app/controllers/api/v1/merchants_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::V1::MerchantsController < ApplicationController 
+
+  def index
+    render json: MerchantSerializer.new(Merchant.all)
+  end
+
+  def show
+    if params[:item_id]
+      render json: MerchantSerializer.new(Item.find(params[:item_id]).merchant)
+    else 
+      render json: MerchantSerializer.new(Merchant.find(params[:id]))
+    end
```

Comment:
returns the `item`'s `merchant` if an `item_id` is passed in the params or the `merchant` otherwise